scanner/src/main/kotlin/storages/PostgresStorage.kt

```diff
@@ -192,11 +192,26 @@ class PostgresStorage(private val connection: Connection, private val schema: St
         // TODO: Check if there is already a matching entry for this provenance and scanner details.
         val query = "INSERT INTO $schema.$table (identifier, scan_result) VALUES (?, to_json(?::json)::jsonb)"
 
+        val scanResultJson = jsonMapper.writeValueAsString(scanResult).escapeNull()
+
         val statement = connection.prepareStatement(query)
         statement.setString(1, id.toCoordinates())
-        statement.setString(2, jsonMapper.writeValueAsString(scanResult))
+        statement.setString(2, scanResultJson)
         statement.execute()
 
         return true
     }
+
+    /**
+     * The null character "\u0000" can appear in raw scan results, for example in ScanCode if the matched text for a
+     * license or copyright contains this character. Since it is not allowed in PostgreSQL JSONB columns we need to
+     * escape it before writing a string to the database.
+     * See: [https://www.postgresql.org/docs/11/datatype-json.html]
+     */
+    private fun String.escapeNull() = replace("\\u0000", "\\\\u0000")
```

Comment:
consider extension functions instead:

fun ResultSet.readStringUnescapedNull(int)
fun JsonMapper.readScanResult()

Reply:
Good idea, will do that separately.

Reply:
I have to say I like the symmetry of the current `String` extension functions better. For me this can stay as-is.